pizza_cutter/des_pizza_cutter/_piff_tools.py

```diff
@@ -0,0 +1,151 @@
+import os
+from functools import lru_cache
+import logging
+import numpy as np
+import piff
+import fitsio
+from ._constants import PSF_IN_BLACKLIST
+
+logger = logging.getLogger(__name__)
+
+
+def load_piff_from_image_path(*, image_path, piff_run):
+    """
+    load a piff object based on the input image path
+
+    Parameters
+    ----------
+    image_path: str
+        A path to an immask file
+    piff_run: str
+        e.g. y3a1-v29
+
+    Returns
+    -------
+    A dict
+        {'psf': piff.PSF,
+         'flags': int,
+         'psf_path': str}
+    """
+
+    paths = _get_paths_from_image_path(image_path, piff_run)
+
+    expinfo = _get_info(paths['info_path'])
+    expnum, ccdnum = _extract_expnum_and_ccdnum(image_path)
+
+    w, = np.where(expinfo['ccdnum'] == ccdnum)
+    if w.size == 0:
+        raise RuntimeError('piff info for exp %s ccd %s '
+                           'not found' % (expnum, ccdnum))
+
+    this_info = expinfo[w[0]]
+
+    piff_flags = 0
+    if not _check_and_log(this_info):
+        piff_flags |= PSF_IN_BLACKLIST
+        psf = None
+    else:
+        psf = _get_piff_psf(paths['psf_path'])
+
+    return {
+        'psf': psf,
+        'flags': piff_flags,
+        'psf_path': paths['psf_path'],
+    }
+
+
+def _check_and_log(info):
+    """
+    check the exp/ccd are OK based on flags and always skipping
+    ccd 31
+    """
+    expnum, ccdnum = info['expnum'], info['ccdnum']
+
+    ok = True
+
+    if info['flag'] != 0:
+        logger.info('skipping bad psf solution for exp %s '
+                    'ccd %s: %s' % (expnum, ccdnum, info['flag']))
+        ok = False
+
+    if info['ccdnum'] == 31:
+        logger.info('skipping ccd 31 for exp %s' % expnum)
+        ok = False
+
+    return ok
+
+
+@lru_cache(maxsize=128)
```

Comment:
What is the memory footprint of each Piff object? We should make sure the cache is not too big.

Reply:
the piff files are 350k and the summary file is about 2M

Reply:
Mmmmk. This is probably fine then.